.vscode/tasks.json

```
@@ -1,58 +1,43 @@
+
 {
   "tasks": [
     {
+      "label": "WaitForDebugpy",
       "type": "shell",
-      "label": "conda: activate environment",
-      "command": "activate freecad",
+      "command": "python ${workspaceFolder}/.vscode/WaitForPort.py",
+      "group": "none",
       "problemMatcher": [],
-      "detail": "Activate conda environment",
-      "group": "build"
+      "presentation": {
+        "reveal": "never", //silently fail and don't launch the debugger
+        "panel": "dedicated",
+        "close": true,
+        "revealProblems": "never"
+      },
+      "hide": true
     },
     {
-      "type": "cmake",
+      "type": "shell",
       "label": "CMake: configure",
-      "command": "configure",
+      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { activate freecad };             cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
+      "linux":   { "command": "if echo \"${command:cmake.activeConfigurePresetName}\" | grep -iqF \"conda\"; then source activate freecad; fi; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
```

```suggestion
      "linux":   { "command": "if echo \"${command:cmake.activeConfigurePresetName}\" | grep -iqF \"conda\"; then source ${workspaceFolder}/.conda/freecad/bin/activate; fi; cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
```
not tested yet but I suspect this would work better, I'll test later and confirm.

---

Should we replace the command with a pair of scripts similar to what you did for windows?